### PR TITLE
Add local port for the client

### PR DESCRIPTION
**Pull request by keepsimple1**
I am trying to run a QUIC client bind with a certain port and found there is no such option today. Hence submitting this PR. Please let me know if it makes sense.  Thank you for your work.

Unit test and manual test are done.

### CODE REVIEW

File: src/aioquic/asyncio/client.py [hunk 1]
```diff
@@ -21,6 +21,7 @@
     create_protocol: Optional[Callable] = QuicConnectionProtocol,
     session_ticket_handler: Optional[SessionTicketHandler] = None,
     stream_handler: Optional[QuicStreamHandler] = None,
+    local_port: Optional[int] = None,
```
Comment (jlaine) [inline src/aioquic/asyncio/client.py]: The variable name is fine, could we have a docstring for it please?

File: .gitignore [hunk 2]
```diff
@@ -8,3 +8,5 @@
 /build
 /dist
 /docs/_build
+.idea
```
Comment (jlaine) [inline .gitignore]: Maybe keep this out of the PR as it's not related

File: src/aioquic/asyncio/client.py (hunk 1 above)
Comment (jlaine) [reply src/aioquic/asyncio/client.py]: I don't think making it `Optional` is right, couldn't we initialize it straight to `0`?

File: tests/test_asyncio.py [hunk 3]
```diff
@@ -287,6 +287,11 @@ def test_connect_timeout(self):
                 )
             )
 
+    def test_connect_local_port(self):
```
Comment (jlaine) [inline tests/test_asyncio.py]: Thank you taking the time to write a test, it makes we want to get this merged :)

**Comment by jlaine** [comment]
Do you still plan to complete this PR with the requested changes?

**Comment by jlaine** [comment]
Could you please remind me what the motivation for this PR is?

File: src/aioquic/asyncio/client.py (hunk 1 above)
Comment (keepsimple1) [reply src/aioquic/asyncio/client.py]: done.

File: .gitignore (hunk 2 above)
Comment (keepsimple1) [reply .gitignore]: done.

**Comment by keepsimple1** [comment]
I apologize about my lengthy delay.  I've updated my diff.  My motivation is that: I was using the client to test a NAT-traversal case, where I need the client to let the server know (via signaling) its port before setting up the QUIC connection.  In such cases, it is quite useful to let client to bind to a particular port.

File: examples/http3_client.py [hunk 4]
```diff
@@ -396,6 +398,13 @@ def save_session_ticket(ticket: SessionTicket) -> None:
     parser.add_argument(
         "-v", "--verbose", action="store_true", help="increase logging verbosity"
     )
+    parser.add_argument(
+        "-lp",
```
Comment (jlaine) [inline examples/http3_client.py:402]: I'm not sure the short version gains us anything, maybe drop it?

Comment (keepsimple1) [reply examples/http3_client.py]: I've dropped it. Personally I liked it, but it's not a must-have, and can be added later if needed.

**Comment by jlaine** [comment]
Could you fix the linter error please?

**Comment by keepsimple1** [comment]
Sure, the linter error is fixed now.

**Comment by keepsimple1** [comment]
Thank you for merging!

**Pull request merged**